### PR TITLE
Relationship practice

### PR DESCRIPTION
**Pull request by axnish27**
implemented a simple CRUD for companies

implemented below relationships
company -> employee  -- 1 - N 
employee ->company -- N - 1
company -> project  -- 1 - N 
project ->company -- N - 1
project -> employee  -- N - N (created a through table)
employee -> BankAccount -- 1 - 1 

@Raza9798

### CODE REVIEW

**Comment by Raza9798** [comment]
can you draw a simple ERD diagram so I can understand easily @axnish27

**Comment by axnish27** [comment]
> can you draw a simple ERD diagram so I can understand easily @axnish27

Sure

**Comment by axnish27** [comment]
image

As requested drew a simple erd to ilustrate the relationship between model please review

File: app/Models/BankAccount.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+
+class BankAccount extends Model
+{
+    use HasFactory;
+
+    public function user(): BelongsTo
```
Comment (Raza9798) [inline app/Models/BankAccount.php:13]: `Employee & Users` are different tables right

File: app/Models/Employee.php [hunk 2]
```diff
@@ -16,5 +19,21 @@ class Employee extends Model
...
+    {
+        return $this->belongsTo(Company::class);
+    }
+
+    public function projects(): BelongsToMany
+    {
+        return $this->belongsToMany(Project::class);
+    }
+
+    public function phone(): HasOne
```
Comment (Raza9798) [inline app/Models/Employee.php:35]: fix the relationship naming

**Review by Raza9798 — CHANGES_REQUESTED** [review]
I found several relationship names which are not properly mentioned. please fix it

File: app/Models/BankAccount.php (hunk 1 above)
Comment (axnish27) [reply app/Models/BankAccount.php:13]: Yess ill fix them..

File: app/Models/Employee.php (hunk 2 above)
Comment (axnish27) [reply app/Models/Employee.php:35]: relation naming fixed #28  @Raza9798  please review again

Comment (Raza9798) [reply app/Models/Employee.php:35]: have you push the changes? i dont see any changes @axnish27

Comment (axnish27) [reply app/Models/Employee.php]: sorry seems like i ddnt...Now pushed please check again

**Comment by Raza9798** [comment]
image

i am not able to see the relationship information in the datatable

**Comment by axnish27** [comment]
> image
> 
> i am not able to see the relationship information in the datatable



I didnt make any changes to the front end or controllers cuz at the time i created a new branch the development branch was way behind and in other branch i had made so many changes to employee controller and view it was pending pull req.....so making any changes to view and controller could have created a big merge conflict...to avoid that i implemented only the backend logic....and very basic crud for companies to that you can find the link nav bar.....


@Raza9798

**Pull request merged**